core/blockchain/indexer.hpp

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_BLOCKCHAIN_INDEXER_HPP
+#define KAGOME_BLOCKCHAIN_INDEXER_HPP
+
+#include <boost/endian/conversion.hpp>
+
+#include "blockchain/block_tree.hpp"
+#include "storage/buffer_map_types.hpp"
+
+namespace kagome::blockchain {
+  struct Descent {
+    Descent(std::shared_ptr<blockchain::BlockTree> block_tree,
+            primitives::BlockInfo start)
+        : block_tree_{std::move(block_tree)}, path_{start} {}
+
+    bool can(const primitives::BlockInfo &to) {
```

Comment:
It looks as if it were a constant method that just checks something, but it's not. Please give it a more descriptive name.